### ISSUE TITLE
Allow other apps to query Tranquille

### ISSUE THREAD

**Issue by Penaz91** [Vinetos/Tranquille#19]
This is just an idea that dawned on me yesterday after receiving a scam SMS (Broken Phone/Hi mom scam): it would be nice if Tranquille allowed a set of applications to interrogate its database for possible spam/scam numbers.

I know for a fact that "ShouldIAnswer" (or "SIA?") supports reporting numbers for spam/scam SMS, so it would be interesting to see integration from other software (like [QUIK](https://github.com/octoshrimpy/quik) or [Fossify Messages](https://github.com/FossifyOrg/Messages)) with Tranquille.

For security and privacy reasons, this should be done via an allow-list kept inside Tranquille, much like [Breezy Weather](https://github.com/breezy-weather/breezy-weather) does with the "Send Data to other apps" functionality.

**Comment by papjul** [comment]
The spam database is not implemented in Tranquille directly. You can integrate it to your project by implementing this library: https://gitlab.com/xynngh/LibPhoneNumberInfo

```kt
[versions]
lib-phone-number-info = "c77bdd2b69"

// ...

[libraries]
lib-phone-number-info = { group = "com.gitlab.xynngh", name = "LibPhoneNumberInfo", version.ref = "lib-phone-number-info" }
```

**Comment by Penaz91** [comment]
That is true, but implementing it would mean having two offline databases that are essentially the same (one for Tranquille and one for the SMS app), hence the idea of allowing apps to go through Tranquille to be able to avoid such duplication.

I'm not an android developer and I imagine there are some limitations in communicating between apps, but I felt like throwing the idea out there to see if it can be done.

**Comment by papjul** [comment]
Yes, the broadcast method used by Breezy Weather is subject to a 1 MB limit. It is starting to cause us [issues](https://github.com/breezy-weather/breezy-weather/issues/1349). We are looking into content providers, which are also limited to 1 MB, but per row, so it will be safe.

To get back to Tranquille, what are you trying to achieve? I know that starting from Android 7, there is a common system block list that can be used by dialer and messaging apps. I don't know if it would be a good thing to store spam numbers there, and if the blocked list is designed to support such a long list. (Note that Tranquille is forked from a very old app that wasn't designed to use that list for the manually input "denylist", but modern apps like Fossify use it, for example)

**Comment by Penaz91** [comment]
The final objective would be having SMS apps warn you about possible spam through a local database (that is not just Android's default deny list) while saving on duplicates between caller-id apps (Tranquille) and the SMS apps.

Not sure if there are other use cases 🤔 

But in the end, if this is not possible (or turns out to be a headache) then it's all good :)
